docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,48 @@
+= BSF Detector
+
+The BSF Detector lets you run arbitrary scripts using the Bean Scripting Framework to determine the availability of a service. This can be a script written in any of the languages supported by BSF, the only requirement is that the script returns the string `OK` if it passes.
+
+For information on writing a BSF script for OpenNMS, see the BSF Monitor page.
+
+Unlike the BSF Monitor, the BSF Detector only passes in the `map`, `ip_addr`, `svc_name`, and `results` variables as beans to the script.
+
+== Detector facts
+
+[options="autowidth"]
+|===
+| Implementation | `org.opennms.netmgt.provision.detector.bsf.BSFDetector`
+|===
+
+== Configuration and Use
+
+.Parameters for the BSF detector
+[options="header, autowidth"]
+
+|===
+| Parameter      | Description                                                                               | Required | Default value
+| `fileName` | The path to the file on the OpenNMS server which contains the BSF-compatible script to run.                                                                        | optional |
+| `langClass`   | The short name of the language the script is written in.
+    Choices: javascript, beanshell, groovy, etc.  | optional | 
+| `bsfEngine`         a| The BSF engine to use when running the script. Choices:
+
+* bsh.util.BeanShellBSFEngine
+* org.codehaus.groovy.bsf.GroovyEngine                | optional | 
+| `fileExtensions`          | A comma-separated list of file extensions to consider a valid script for this BSF instance.                                                                              | optional | 
+| `runType`      a| Which way to run the BSF script. Choices:
+
+* `eval` - expects the script to print the OK/error status to
+* `exec` - expects the script to manipulate the provided results hashmap directly                                         | optional | `eval`
+|===
+
```

Comment:
all the empty new lines here

Can we provide a simple BeanShell script as a working example?

Reply:
> all the empty new lines here

The empty lines are required to get the list here rendered like this:
![image](https://user-images.githubusercontent.com/4995222/111802606-7a370a00-88ce-11eb-94e5-f73da34a9ad6.png)

Without that it looks like a list is not doable. Does it break any tests?

> Can we provide a simple BeanShell script as a working example?

:shrug: I was hoping you guys have one ;-) But of course we can create one.

Reply:
I think you can create an example from the https://github.com/OpenNMS/opennms/blob/develop/opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/BSFMonitor.adoc.